packages/editor/src/browser/editor-widget.ts

```diff
@@ -32,6 +34,14 @@ export class EditorWidget extends BaseWidget implements SaveableSource, Navigata
                 this.selectionService.selection = this.editor;
             }
         }));
+        this.toDispose.push(this.labelProvider.onDidChange(async uri => {
+            const resourceUri = this.editor.getResourceUri();
+            if (resourceUri && resourceUri.toString() === uri.toString()) {
```

Comment:
`resourceUri` and `uri` in `editor-widget-factory`, it is not the same

i.e. `uri` can diff uri but `resourceUr` points to a file for which diff is computed

if we update here based on `resourceUri` then it will break captions for diff editors